### PR TITLE
add tests

### PR DESCRIPTION
**Pull request by SilverBzH**
- pica python: make it as a standalone package
- test: add pytest and implement a test runner

### CODE REVIEW

File: .github/workflows/python_format.yml [hunk 1]
```diff
@@ -20,4 +20,4 @@ jobs:
         run: |
           pip install --upgrade pip
           pip install black=="23.12.1"
-      - run: black --check scripts/ --exclude pica
+      - run: black --check tests/
```
Comment (hchataing) [inline .github/workflows/python_format.yml:23]: `py/pica` as well

File: README.md [hunk 2]
```diff
@@ -117,3 +117,22 @@ $> --> pica_create_anchor 00:01 # Create another one
...
+
+# Tests
+
+setup your python env:
+
+```bash
+python3 -m venv venv
+source venv/bin/activate
+pip install pytest
+pip install -e py/pica/
```
Comment (hchataing) [inline README.md:129]: Missing `pytest_asyncio`

File: README.md [hunk 3]
```diff
@@ -117,3 +117,22 @@ $> --> pica_create_anchor 00:01 # Create another one
...
+```
+
+Then run the Tests
```
Comment (hchataing) [inline README.md:132]: nit: `Then run the tests`

File: README.md [hunk 4]
```diff
@@ -117,3 +117,22 @@ $> --> pica_create_anchor 00:01 # Create another one
...
+
+```bash
+pytest --log-cli-level=DEBUG -v
+```
+
+The tests are located at `./tests/`
```
Comment (hchataing) [inline README.md:138]: nit: `located in`

File: py/pica/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+[project]
+name = "pica"
+dynamic = ["version"]
+description = "pica python uci packets"
```
Comment (hchataing) [inline py/pica/pyproject.toml:4]: nit: `UCI host helpers` ?
(there is also the serial port management)

File: tests/console.py
Comment (hchataing) [inline tests/console.py:1]: The console is not really a test; should we keep it within `py/pica` ?

File: tests/ranging.py [hunk 6]
```diff
@@ -234,12 +235,12 @@ async def controlee(host: Host, peer: Host):
     await host.expect_control(uci.SessionDeinitRsp(status=uci.StatusCode.UCI_STATUS_OK))
 
 
-async def run(address: str, uci_port: int, http_port: int):
+async def run(address: str, uci_port: int):
```
Comment (hchataing) [inline tests/ranging.py:238]: Please remove the command line option as well
(forgot to clean that up at the time).

File: tests/data_transfer.py [hunk 7]
```diff
@@ -158,7 +161,7 @@ async def data_transfer(host: Host, dst_mac_address: bytes, file: str, session_i
         print(e)
 
 
-async def run(address: str, uci_port: int, http_port: int, file: str):
+async def run(address: str, uci_port: int, file: Path):
```
Comment (hchataing) [inline tests/data_transfer.py:164]: Please remove the command line option as well
(forgot to clean that up at the time).

File: tests/console.py
Comment (SilverBzH) [reply tests/console.py:1]: Agree

**Review by hchataing — APPROVED** [review]

**Pull request merged**